### PR TITLE
fix: Sorting icon for table update [MDS-971]

### PR DESCRIPTION
**Pull request by salvatorecriscioneweb**
### Issue:

Sorting icon is wrong, we need to use by default `chevron-small`, plus I added the possibility to customize it in case needed.

### API Changes:

- Is possible to customize the color using the `text-*` class, `sorting_icon_color` is added to cover it, by default is set to `bulma` for `Moon.Parts` , in the design is unset
- Is possible to customize the icon using the `sorting_icon` prop, by default `controls_chevron_up_small`
note:
You need to specify only the `up` icon, the `down` icon is rendered by rotating the up by 180deg

### Default:

Screenshot 2024-01-23 at 12 06 42

### Example:

sorting-icon

### CODE REVIEW

File: lib/moon/design/table.ex
Comment (phgrey) [inline lib/moon/design/table.ex:1]: for me it looks like breaking change: default icon for Design.Table changes arrows_up -> controls_chevron_up_small

Comment (salvatorecriscioneweb) [reply lib/moon/design/table.ex:1]: Recover to `arrows_up` for the Design, and leave `chevron` only for Parts? maybe can be a solution

Comment (phgrey) [reply lib/moon/design/table.ex:1]: Table is widely used and we don't like notifications aka "table is broken" in support channels.

Comment (salvatorecriscioneweb) [reply lib/moon/design/table.ex:1]: @phgrey I received confirmation from the Design team, and then we have the green light to use the `chevron` icon. The use of the `arrow_up` icon with Moon is incorrect. 💯 

Certainly, it's essential to notify consumers of the changes to the default sorting icon. If they want to have `arrow_up`, they should still have the option to customize to it.

**Pull request merged**